File: Data_Structure_Topics/Insertion_Sort.js



const myArr = [2, 3, 6, 4, 1, 7];

var temp, j;

for (let i = 1; i < myArr.length; i++) {
    temp = myArr[i];
    j = i - 1;
    while (j >= 0 && myArr[j] > temp) {
        myArr[j + 1] = myArr[j];
        j--;
    }
    myArr[j + 1] = temp;
}

console.log(myArr);

File: Data_Structure_Topics/Insertion_Sort.py




myArr = [2, 1, 5, 3, 7, 6, 9]
l = len(myArr)  
for one in range(1, l):
    temp = myArr[one]
    j = one - 1
    while j >= 0 and myArr[j] > temp:
        myArr[j + 1] = myArr[j]
        j -= 1
    myArr[j + 1] = temp

print(myArr)
